README.md

```diff
@@ -1,17 +1,103 @@
+
+## STM32flash and DFU Mode
+### Requirements
+
+* Download the .hex file you want to flash e.g. look at the [releases section](https://github.com/Nitrokey/nitrokey-pro-firmware/releases) or build it yourself (see above).
+* You may use [STM32 Flash Loader Demonstrator](http://www.st.com/content/st_com/en/products/development-tools/software-development-tools/stm32-software-development-tools/stm32-programmers/flasher-stm32.html) (Windows only) or the open source command line tool [stm32flash](http://stm32flash.sourceforge.net). *Note: the terminal commands below are based on the command line tool.*
+* If your computer doesn't has a RS232 port (most modern laptops don't have it) you would need a USB-to-RS232/TTL adapter. [Sparkfun BOB-00718](https://www.sparkfun.com/products/718) should work (untested) and you can find even cheaper adapters online. Previously we built [our own adapter](https://www.assembla.com/spaces/cryptostick/wiki/Crypto_Stick_Programmer) which hardware layout you can [download](https://www.assembla.com/spaces/cryptostick/subversion/source/HEAD/trunk/hardware/programmer%20for%20v1.2).
+* You would need a simple USB adapter to bridge Nitrokey's USB plug to the USB-to-RS232 adapter.
+
+Your adapter should consist of a USB socket which four pins are connected to your serial/TTL connector. The pinout is as follows.
+
+```
+Nitrokey USB Plug <-> Serial/TTL adapter
+
+Pin 1, VCC <-> VCC
+Pin 2, D-  <-> TX
+Pin 3, D+  <-> RX
+Pin 4, GND <-> GND
+
+This diagram represents the pinout of the USB socket which you are going to solder:
+
+  ################### 
+  #                 # 
+  # ############### # 
+  #                 # 
+  #                 # 
+  ################### 
+     #   #   #   #   
+     #   #   #   #    
+
+     1   2   3   4
+```
+
+The following picture shows the adapter/USB-to-TTL connection.
+![USB-to-TTL adapter and USB socket](adapter.jpg?raw=true)
+
+To flash the firmware you need to bridge the two contact holes and only then connect (and power) the PCB to your adapter. The bridge triggers the hardware to boot into DFU mode. You can use a jumper with 2.0 mm pitch or just prepare/solder a wire. The following picture shows a bridge for the Nitrokey.
+
+![Nitrokey bridged with a jumper](connected_jumper.jpg?raw=true)
+
+### Actual Flashing 
+
+While the jumper is plugged in, connect the Nitrokey to the USB-serial adapter on your computer. The jumper is only required during the first moment of connection and can be removed afterwards.
+
+You can check if the Nitrokey got successfully into DFU mode by typing in the following into a terminal:
+
+```
+$ sudo stm32flash /dev/ttyUSB0
+
+stm32flash 0.5
+
+http://stm32flash.sourceforge.net/
+
+Error probing interface "serial_posix"
+Cannot handle device "/dev/ttyUSB0"
+Failed to open port: /dev/ttyUSB0
```

Comment:
Should not here be any success message? :-)

Reply:
I try to test it again these days... maybe I took wrong output -.-

Reply:
As this is deprecated anyway I won't further test.